### PR TITLE
Add support for openocd on stm32g0 and stm32g4 targets

### LINKED ISSUE #24055: Add support for openocd on stm32g0 and stm32g4 targets
**Is your enhancement proposal related to a problem? Please describe.**
stm32g0 and stm32g4 soc support has recently been added on openocd.
openocd support can now be enabled on these targets when flashing  debugging

**Describe the solution you'd like**
Identify openocd SHA1 that enable G4/G0 support.
Enable openocd support on zephyr G4 and G0 supported platforms.
Update board docs
Check everything is fine flash/debug
Request needed change on https://github.com/zephyrproject-rtos/openocd
Close related issues (https://github.com/zephyrproject-rtos/openocd/pull/11)

**Additional context**
Will avoid need to update pyocd local version to support G4 packages.

### PR DESCRIPTION
**Pull request by FRASTM**
This patch _plus_ the update of the openocd https://github.com/zephyrproject-rtos/openocd/issues/24
will add the support of stm32g0 and stm32g4 soc with openocd.

In case a local version of the openocd which supports the G0 and G4, is required, refer to the issue mentionned below to install it and use the `west flash --openocd /usr/local/bin/openocd `

Fix https://github.com/zephyrproject-rtos/zephyr/issues/24055

Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

File: boards/arm/nucleo_g071rb/board.cmake [hunk 1]
```diff
@@ -3,4 +3,6 @@ board_runner_args(pyocd "--target=stm32g071rb")
 board_runner_args(pyocd "--flash-opt=-O reset_type=hw")
 board_runner_args(pyocd "--flash-opt=-O connect_mode=under-reset")
 
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```
Comment (erwango) [inline boards/arm/nucleo_g071rb/board.cmake:6]: Please put in second position for now.

**Review by erwango — CHANGES_REQUESTED** [review]
Since G0/G4 openocd support is not yet available in zephyr sdk, please make the situation clear in commit messages and board docs by adding a pointer to openocd repo and match commit.

File: boards/arm/nucleo_g071rb/board.cmake (hunk 1 above)
Comment (FRASTM) [reply boards/arm/nucleo_g071rb/board.cmake]: done

**Comment by FRASTM** [comment]
put runner for the openocd in second position in each `board.cmake` file

**Comment by FRASTM** [comment]
Updating the doc files to mention the _local_ openocd

**Comment by FRASTM** [comment]
add details in the PR itself

File: boards/arm/nucleo_g071rb/doc/index.rst [hunk 2]
```diff
@@ -133,7 +133,16 @@ Flashing
...
+You can still update, compile and install a 'local' openocd from the official repo http://openocd.zylin.com
+and run the openocd command:
+
+.. code-block:: console
+
+   $ west flash --openocd /usr/local/bin/openocd
+
+Note that '/usr/local/bin/openocd' is your path for the freshly installed openocd, given by "$ which openocd"
+
 Instead, support can be enabled on pyocd by adding "pack" support with
```
Comment (erwango) [inline boards/arm/nucleo_g071rb/doc/index.rst:146]: Can you rework a bit to have a clean transitin with the next paragraph about current pyocd support?

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_g071rb/doc/index.rst (hunk 2 above)
Comment (FRASTM) [reply boards/arm/nucleo_g071rb/doc/index.rst:146]: changing text to explain the use of openocd interface

**Comment by FRASTM** [comment]
changing text to explain the use of openocd interface

**Comment by FRASTM** [comment]
refine doc/index.rst

**Review by erwango — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**